tntconcept-core/src/main/java/com/autentia/tnt/manager/holiday/UserHolidaysStateManager.java

```diff
@@ -41,221 +45,228 @@
+			if (DateUtils.isSameDay(current.getDate(), date)) {
 				return true;
 			}
 		}
-		
+
 		return false;
 	}
-	
-	
-  	/**
-  	 * @return Devuelve el número de días laborables que hay entre dos fechas
-  	 */
-	public int getWorkingDays(Date fromDate, Date toDate) {
-		int			  total			= 0;
+
+	/**
+	 * @return Devuelve el número de días laborables que hay entre dos fechas
+	 */
+	public int getWorkingDays(User user, Date fromDate, Date toDate) {
+		int total = 0;
 
 		// Evitamos un bucle infinito en el bucle que viene a continuación
-		if (fromDate.before(toDate) || DateUtils.isSameDay(fromDate, toDate)){
-			HolidaySearch fiestaSearch  = new HolidaySearch();
-			Date 		   current  	= (Date) fromDate.clone();
-			
+		if (fromDate.before(toDate) || DateUtils.isSameDay(fromDate, toDate)) {
+			HolidaySearch fiestaSearch = new HolidaySearch();
+			Date current = (Date) fromDate.clone();
+
 			fiestaSearch.setStartDate(fromDate);
 			fiestaSearch.setEndDate(toDate);
-			List<Holiday> fiestas = HolidayManager.getDefault().getAllEntities(fiestaSearch, null);
-		 
-			
-			while (current.before(toDate) || DateUtils.isSameDay(current, toDate)){
-				if (! this.isHoliday(fiestas, current)){
+
+			CorrespondingHolidayManager correspondingFiestasManager = CorrespondingHolidayManager.getDefault();
+
+			List<Holiday> allFiestas = HolidayManager.getDefault().getAllEntities(fiestaSearch, null);
+
+			Calendar fromYear = Calendar.getInstance();
+			Calendar toYear = Calendar.getInstance();
+			fromYear.setTime(fromDate);
+			toYear.setTime(toDate);
+
+			List<Holiday> correspondingFiestas = correspondingFiestasManager.calcCorrespondingHolidays(fromYear, toYear,
+					allFiestas, user.getStartDate());
+
+			while (current.before(toDate) || DateUtils.isSameDay(current, toDate)) {
+				if (!this.isHoliday(correspondingFiestas, current)) {
 					total++;
 				}
 				current = DateUtils.addDays(current, 1);
-			}			
+			}
 		}
-		
 
-		
 		return total;
 	}
 
 	public UserHolidaysState calcUserHolidaysState(User usuario, Date chargeYear) {
 		UserHolidaysState uhs = new UserHolidaysState();
 		uhs.setUser(usuario);
-		// WorkingAgreement attribute is an HB proxy not initialized, we need to get it from DB
+		// WorkingAgreement attribute is an HB proxy not initialized, we need to get it
+		// from DB
 		WorkingAgreementDAO workingAgreementDao = WorkingAgreementDAO.getDefault();
 		WorkingAgreement agreement = workingAgreementDao.getById(usuario.getAgreement().getId());
 		// and refresh the user instance agreement attribute
 		usuario.setAgreement(agreement);
-		
-		
+
 		uhs.setTotalYear(agreement.getHolidays());
-		
+
 		int acceptedHolidays = 0;
-		
-		if(chargeYear!=null) {
-			
+
+		if (chargeYear != null) {
+
 			HolidayManager fiestasManager = HolidayManager.getDefault();
-			
+
 			// We must take in account previous year holidays and next year holidays
-			
-		Calendar calMin = Calendar.getInstance();
+
+			Calendar calMin = Calendar.getInstance();
```

Comment:
Refactor this code and create a method to create the instance calendar with the year

Reply:
Done